reconcile/templating/lib/rendering.py

```diff
@@ -123,11 +123,31 @@ def render_output(self) -> str:
         )
 
         if isinstance(matched_value, list):
-            if not self.template.patch.identifier:
-                raise ValueError(
-                    f"Expected identifier in patch for list at {self.template}"
-                )
-            dta_identifier = data_to_add.get(self.template.patch.identifier)
+
+            def get_identifier(data: dict[str, Any] | str) -> Any:
+                assert self.template.patch is not None  # mypy
+                if not self.template.patch.identifier:
+                    raise ValueError(
+                        f"Expected identifier in patch for list at {self.template}"
+                    )
+                if isinstance(data, str):
```

Comment:
should you check `identifier == "."` ?

Reply:
`patch.identifier` is not used for lists, but it's mandatory. I think `patch.identifier: .` is just convenient. Added the check in the last commit